code/modules/mob/living/carbon/xenomorph/castes/defiler/abilities_defiler.dm

```diff
@@ -1,65 +1,110 @@
+/datum/action/xeno_action/activable/defile/on_cooldown_finish()
+	owner.playsound_local(owner, 'sound/effects/xeno_newlarva.ogg', 25, 0, 1)
+	to_chat(owner, "<span class='xenodanger'>You feel your toxin glands refill and another young one ready for implantation. You can use Defile again.</span>")
 	return ..()
 
-/datum/action/xeno_action/activable/larval_growth_sting/defiler/use_ability(atom/A)
+/datum/action/xeno_action/activable/defile/can_use_ability(atom/A, silent = FALSE, override_flags)
+	. = ..()
+	if(!.)
+		return
+
+	if(!iscarbon(A))
+		if(!silent)
+			to_chat(owner, "<span class='xenodanger'>Our sting won't affect this target!</span>")
+		return FALSE
+
+	var/mob/living/carbon/victim = A
+
+	if(!A?.can_sting())
+		if(!silent)
+			to_chat(owner, "<span class='xenodanger'>Our sting won't affect this target!</span>")
+		return FALSE
+
+	if(CHECK_BITFIELD(victim.status_flags, XENO_HOST)) // already got one, stops doubling up
+		if(!silent)
+			to_chat(owner, "<span class='xenodanger'>This target already has a young one!</span>")
+		return FALSE
+
+	if(!owner.Adjacent(A))
+		var/mob/living/carbon/xenomorph/X = owner
+		if(!silent)
+			to_chat(X, "<span class='warning'>We can't reach this target!</span>")
+		return FALSE
+
+
+/datum/action/xeno_action/activable/defile/use_ability(atom/A)
 	var/mob/living/carbon/xenomorph/Defiler/X = owner
 	var/mob/living/carbon/C = A
-	if(locate(/obj/item/alien_embryo) in C) // already got one, stops doubling up
-		return ..()
+	X.face_atom(C)
+	X.do_attack_animation(C)
 	if(!do_after(X, DEFILER_STING_CHANNEL_TIME, TRUE, C, BUSY_ICON_HOSTILE))
 		return fail_activate()
 	if(!can_use_ability(A))
 		return fail_activate()
 	add_cooldown()
-	X.face_atom(C)
-	X.do_attack_animation(C)
 	playsound(C, pick('sound/voice/alien_drool1.ogg', 'sound/voice/alien_drool2.ogg'), 15, 1)
+	playsound(C, 'sound/effects/spray3.ogg', 15, TRUE)
 	var/obj/item/alien_embryo/embryo = new(C)
 	embryo.hivenumber = X.hivenumber
 	GLOB.round_statistics.now_pregnant++
+	C.reagents.add_reagent(/datum/reagent/toxin/xeno_growthtoxin, REAGENTS_OVERDOSE, no_overdose = TRUE) //Inject the maximum possible larval growth without ODing
 	SSblackbox.record_feedback("tally", "round_statistics", 1, "now_pregnant")
 	to_chat(X, "<span class='xenodanger'>Our stinger successfully implants a larva into the host.</span>")
-	to_chat(C, "<span class='danger'>You feel horrible pain as something large is forcefully implanted in your thorax.</span>")
-	C.apply_damage(100, STAMINA)
-	C.apply_damage(10, BRUTE, "chest", updating_health = TRUE)
+	to_chat(C, "<span class='highdanger'>You feel horrible pain as something large is forcefully implanted in your thorax.</span>")
+	C.adjust_stagger(3)
+	C.add_slowdown(3)
+	C.adjust_blurriness(3) //Cosmetic eye blur SFX
+	C.reagents.add_reagent(X.selected_reagent, DEFILER_DEFILE_INJECT_AMOUNT, no_overdose = TRUE) //Inject selected chem with OD protection
```

Comment:
i don't see why injecting selected chem is needed, you already have reagent slash and smoke for that.

Reply:
Probably too much with the ability to swap the smoke chem type emitted mid-use, but I've removed that so this is fine given the set up required